crates/valence_protocol/src/var_long.rs

```diff
@@ -26,7 +30,59 @@ impl VarLong {
 }
 
 impl Encode for VarLong {
+    // Adapted from VarInt-Simd encode
+    // https://github.com/as-com/varint-simd/blob/0f468783da8e181929b01b9c6e9f741c1fe09825/src/encode/mod.rs#L71
+    #[cfg(all(
+        any(target_arch = "x86", target_arch = "x86_64"),
+        not(target_os = "macos")
+    ))]
     fn encode(&self, mut w: impl Write) -> Result<()> {
```

Comment:
Looks like we're missing a fallback implementation for non-x86 or non-x86_64 platforms. If you decide to use this approach, please make a generic implementation that works on all platforms.

Reply:
Would this work for the fallback implementation?
```
#[cfg(any(
    not(any(target_arch = "x86", target_arch = "x86_64")),
    target_os = "macos"
))]
```

Reply:
Oh wait I didn't read the rest of the code. You're fine.